[Automation] native-image run fails for info.picocli:picocli:4.7.6

## native-image run on ubuntu-latest / JDK 25 / mode current-defaults

Failure kind: native-image run
OS: ubuntu-latest
JDK: 25
Native-image mode: current-defaults
Reproducer: `GVM_TCK_NATIVE_IMAGE_MODE="current-defaults" GVM_TCK_LV="4.7.6" ./gradlew clean test -Pcoordinates="info.picocli:picocli:4.7.6"`
Runner log: https://github.com/oracle/graalvm-reachability-metadata/actions/runs/25727047580/job/75542581632
Last 300 lines of the log (excerpt truncated to fit the GitHub issue size limit):
```console
3,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/info.picocli/picocli/4.7.6 \
-H:Optimize@user+api=b \
-H:+ReportExceptionStackTraces@user \
-H:Features@user+api=org.graalvm.junit.platform.JUnitPlatformFeature \
'-H:Class@explicit main-class=org.graalvm.junit.platform.NativeImageJUnitLauncher' \
-H:ImageBuildID@driver=de0fee8a-9542-9f95-cbd3-e1849e26a176 \
'-H:Features@jar:file:///opt/hostedtoolcache/graalvm-jdk-25_linux-x64_bin/25.0.0/x64/graalvm-jdk-25.0.3+9.1/lib/svm/library-support.jar!/META-INF/native-image/com.oracle.svm/thirdparty/native-image.properties+api=com.oracle.svm.thirdparty.gson.GsonFeature' \
'-H:Features@jar:file:///opt/hostedtoolcache/graalvm-jdk-25_linux-x64_bin/25.0.0/x64/graalvm-jdk-25.0.3+9.1/lib/svm/library-support.jar!/META-INF/native-image/com.oracle.svm/polyglot/native-image.properties+api=com.oracle.svm.polyglot.groovy.GroovyIndyInterfaceFeature,com.oracle.svm.polyglot.scala.ScalaFeature'
]
Warning: Option 'DynamicProxyConfigurationResources' is deprecated and might be removed in a future release: This can be caused by a proxy-config.json file in your META-INF directory. Consider including proxy configuration in the reflection section of reachability-metadata.md instead.. Please refer to the GraalVM release notes.
Warning: Option 'DynamicProxyConfigurationResources' is deprecated and might be removed in a future release: This can be caused by a proxy-config.json file in your META-INF directory. Consider including proxy configuration in the reflection section of reachability-metadata.md instead.. Please refer to the GraalVM release notes.
========================================================================================================================
GraalVM Native Image: Generating 'info.picocli.picocli_tests-tests' (executable)...
========================================================================================================================
For detailed information and explanations on the build output, visit:
https://github.com/oracle/graal/blob/master/docs/reference-manual/native-image/BuildOutput.md
------------------------------------------------------------------------------------------------------------------------
[1/8] Initializing...                                                                                    (6.1s @ 0.17GB)
 Java version: 25.0.3+9-LTS, vendor version: Oracle GraalVM 25.0.3+9.1
 Graal compiler: optimization level: b, target machine: x86-64-v3, PGO: off
 C compiler: gcc (linux, x86_64, 13.3.0)
 Garbage collector: Serial GC (max heap size: 80% of RAM)
 2 user-specific feature(s):
 - com.oracle.svm.thirdparty.gson.GsonFeature
 - org.graalvm.junit.platform.JUnitPlatformFeature
------------------------------------------------------------------------------------------------------------------------
 1 experimental option(s) unlocked:
 - '-H:BuildOutputJSONFile' (origin(s): command line)
------------------------------------------------------------------------------------------------------------------------
Build resources:
 - 12.67GB of memory (75.6% of system memory, $CI set to 'true')
 - 4 thread(s) (100.0% of 4 available processor(s), determined at start)
[junit-platform-native] Running in 'test listener' mode using files matching pattern [junit-platform-unique-ids*] found in folder [/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/info.picocli/picocli/4.7.6/build/test-results/test/testlist] and its subfolders.
[2/8] Performing analysis...  [*****]                                                                   (36.4s @ 0.87GB)
    8,422 types,   9,692 fields, and  37,973 methods found reachable
    2,809 types,     296 fields, and   1,931 methods registered for reflection
       60 types,      68 fields, and      52 methods registered for JNI access
        0 downcalls and 0 upcalls registered for foreign access
        4 native libraries: dl, pthread, rt, z
[3/8] Building universe...                                                                               (4.4s @ 0.97GB)
[4/8] Parsing methods...      [**]                                                                       (4.2s @ 0.63GB)
[5/8] Inlining methods...     [***]                                                                      (2.4s @ 0.69GB)
[6/8] Compiling methods...    [****]                                                                    (19.0s @ 0.83GB)
[7/8] Laying out methods...   [**]                                                                       (3.4s @ 0.80GB)
[8/8] Creating image...       [**]                                                                       (3.5s @ 0.90GB)
  10.18MB (36.70%) for code area:    23,423 compilation units
  13.63MB (49.13%) for image heap:  218,920 objects and 167 resources
   3.93MB (14.17%) for other data
  27.74MB in total image size, 25.63MB in total file size
------------------------------------------------------------------------------------------------------------------------
Top 10 origins of code area:                                Top 10 object types in image heap:
   4.76MB java.base                                            3.02MB byte[] for code metadata
   2.17MB svm.jar (Native Image)                               2.64MB byte[] for java.lang.String
 614.55kB picocli-4.7.6.jar                                    1.39MB java.lang.String
 582.11kB assertj-core-3.22.0.jar                              1.05MB java.lang.Class
 481.18kB java.xml                                           875.89kB com.oracle.svm.core.hub.DynamicHubCompanion
 208.65kB junit-jupiter-engine-5.9.2.jar                     720.37kB byte[] for general heap data
 121.66kB junit-platform-launcher-1.9.2.jar                  453.02kB java.util.concurrent.ConcurrentHashMap$Node
 103.55kB java.logging                                       358.94kB java.util.HashMap$Node
  99.82kB junit-platform-engine-1.9.2.jar                    290.12kB java.lang.Object[]
  82.01kB com.oracle.svm.svm_enterprise                      245.09kB c.o.svm.core.hub.DynamicHub$ReflectionMetadata
 585.83kB for 27 more packages                                 2.58MB for 2060 more object types
                            Use '--emit build-report' to create a report with more details.
------------------------------------------------------------------------------------------------------------------------
Security report:
 - Binary includes Java deserialization.
 - CycloneDX SBOM with 18 component(s) is embedded in binary (934B). 217 type(s) could not be associated to a component.
 - Advanced obfuscation not enabled; enable with '-H:AdvancedObfuscation=""' (experimental support).
------------------------------------------------------------------------------------------------------------------------
Recommendations:
 G1GC: Use the G1 GC ('--gc=G1') for improved latency and throughput.
 PGO:  Use Profile-Guided Optimizations ('--pgo') for improved throughput.
 FUTR: Use '--future-defaults=all' to prepare for future releases.
 HEAP: Set max heap for improved and more predictable memory usage.
 CPU:  Enable more CPU features with '-march=native' for improved performance.
------------------------------------------------------------------------------------------------------------------------
                        5.0s (6.0% of total time) in 755 GCs | Peak RSS: 1.72GB | CPU load: 3.63
------------------------------------------------------------------------------------------------------------------------
Build artifacts:
 /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/info.picocli/picocli/4.7.6/build/native/nativeTestCompile/info.picocli.picocli_tests-tests (executable)
 /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/info.picocli/picocli/4.7.6/build/native/nativeTestCompile/libmanagement_ext.so (jdk_library)
 /tmp/native-image-build-output.json (build_info)
========================================================================================================================
Finished generating 'info.picocli.picocli_tests-tests' in 1m 21s.
[native-image-plugin] Native Image written to: /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/info.picocli/picocli/4.7.6/build/native/nativeTestCompile
WARNING: A terminally deprecated method in sun.misc.Unsafe has been called
WARNING: sun.misc.Unsafe::objectFieldOffset has been called by info_picocli.picocli.CommandLineInnerInterpreterTest (file:/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/info.picocli/picocli/4.7.6/build/native/nativeTestCompile/info.picocli.picocli_tests-tests)
WARNING: Please consider reporting this to the maintainers of class info_picocli.picocli.CommandLineInnerInterpreterTest
WARNING: sun.misc.Unsafe::objectFieldOffset will be removed in a future release

> Task :nativeTest FAILED
[junit-platform-native] WARNING: -djunit.platform.listeners.uid.tracking.output.dir not specified, trying to find test-ids on default Gradle/Maven locations. As this is a fallback mode, it could take a while. This should only happen if you are running tests executable manually and you didn't pass uid output directory with -djunit.platform.listeners.uid.tracking.output.dir=<path>.
[junit-platform-native] WARNING: Using test-ids from default Gradle project location:/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/info.picocli/picocli/4.7.6/build/test-results/test/testlist
[junit-platform-native] Running in 'test listener' mode using files matching pattern [junit-platform-unique-ids*] found in folder [/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/info.picocli/picocli/4.7.6/build/test-results/test/testlist] and its subfolders.
JUnit Platform on Native Image - report
----------------------------------------

groovy.lang.CommandLineInnerDefaultFactoryTest > invokesGroovyClosureConstructorAndCallThroughDefaultFactory() SUCCESSFUL

groovy.lang.CommandLineInnerDefaultFactoryTest > createsPublicNoArgInstancesWithDefaultFactory() SUCCESSFUL

groovy.lang.CommandLineInnerDefaultFactoryTest > createsInstancesWithPrivateNoArgConstructors() SUCCESSFUL

groovy.lang.CommandLineInnerDefaultFactoryTest > invokesDeclaredConstructorWhenPublicInstantiationFails() SUCCESSFUL

groovy.lang.CommandLineInnerDefaultFactoryTest > createsInstanceWhenPublicConstructorSucceedsOnDeclaredRetry() SUCCESSFUL

info_picocli.picocli.AutoCompleteInnerAppTest > generatesCompletionScriptFromNamedCommandAndFactoryClasses() SUCCESSFUL

info_picocli.picocli.CommandLineInnerBuiltInInnerClassConverterTest > convertsFullyQualifiedClassNameOptionToClass() SUCCESSFUL

info_picocli.picocli.CommandLineInnerBuiltInInnerISO8601TimeConverterTest > convertsIso8601TimeOptionsToSqlTime() SUCCESSFUL

info_picocli.picocli.CommandLineInnerBuiltInInnerReflectionConverterTest > convertsBuiltInTypesUsingReflectionBackedConverters() SUCCESSFUL

info_picocli.picocli.CommandLineInnerHelpInnerAnsiTest > autoAnsiEvaluatesTerminalWhenTtyModeIsRequested() SUCCESSFUL

info_picocli.picocli.CommandLineInnerHelpInnerAnsiTest > autoAnsiEvaluatesConsoleTerminalInPseudoTerminalProcess() FAILED

info_picocli.picocli.CommandLineInnerInterpreterTest > expandsArrayOptionValues() SUCCESSFUL

info_picocli.picocli.CommandLineInnerInterpreterTest > togglesOptionalBooleanFromInitialValue() SUCCESSFUL

info_picocli.picocli.CommandLineInnerInterpreterTest > readsInteractivePasswordWithoutEcho() SUCCESSFUL

info_picocli.picocli.CommandLineInnerModelInnerArgGroupSpecTest > appendsRepeatedArgGroupInstancesToArrayField() SUCCESSFUL

info_picocli.picocli.CommandLineInnerModelInnerCommandSpecTest > resourceBundleBaseNameLoadsBundleAndUpdatesCommandArguments() SUCCESSFUL

info_picocli.picocli.CommandLineInnerModelInnerCommandUserObjectTest > populateSpecCreatesProxyForAnnotatedInterfaceCommand() SUCCESSFUL

info_picocli.picocli.CommandLineInnerModelInnerMessagesTest > loadsResourceBundleByBaseName() SUCCESSFUL

info_picocli.picocli.CommandLineInnerModelInnerMessagesTest > acceptsExistingResourceBundle() SUCCESSFUL

info_picocli.picocli.CommandLineInnerModelInnerMethodBindingTest > parseArgsInvokesAnnotatedSetterMethods() SUCCESSFUL

info_picocli.picocli.CommandLineInnerModelInnerUsageMessageSpecAnonymous1Test > autoUsageWidthDetectsTerminalInPseudoTerminalProcess() FAILED

info_picocli.picocli.CommandLineInnerPropertiesDefaultProviderTest > loadsDefaultsFromClasspathResourceWhenUserHomeFileIsAbsent(Path) SUCCESSFUL

info_picocli.picocli.CommandLineTest > executesStaticCommandMethod() SUCCESSFUL

info_picocli.picocli.CommandLineTest > mapsOptionalMethodOptionsToEmptyAndPresentValues() SUCCESSFUL

info_picocli.picocli.CommandLineTest > executesInstanceCommandMethodAsSubcommand() SUCCESSFUL

info_picocli.picocli.CommandLineTest > findsPublicInheritedAndDeclaredCommandMethods() SUCCESSFUL


Failures (2):
  JUnit Jupiter:CommandLineInnerHelpInnerAnsiTest:autoAnsiEvaluatesConsoleTerminalInPseudoTerminalProcess()
    MethodSource [className = 'info_picocli.picocli.CommandLineInnerHelpInnerAnsiTest', methodName = 'autoAnsiEvaluatesConsoleTerminalInPseudoTerminalProcess', methodParameterTypes = '']
    => java.lang.NullPointerException
       java.base@25.0.3/java.util.Objects.requireNonNull(Objects.java:220)
       java.base@25.0.3/sun.nio.fs.UnixFileSystem.getPath(UnixFileSystem.java:257)
       java.base@25.0.3/java.nio.file.Path.of(Path.java:148)
       java.base@25.0.3/java.nio.file.Paths.get(Paths.java:69)
       info_picocli.picocli.CommandLineInnerHelpInnerAnsiTest.runChildProcessInPseudoTerminal(CommandLineInnerHelpInnerAnsiTest.java:77)
       info_picocli.picocli.CommandLineInnerHelpInnerAnsiTest.autoAnsiEvaluatesConsoleTerminalInPseudoTerminalProcess(CommandLineInnerHelpInnerAnsiTest.java:48)
       java.base@25.0.3/java.lang.reflect.Method.invoke(Method.java:565)
       org.junit.platform.commons.util.ReflectionUtils.invokeMethod(ReflectionUtils.java:727)
       org.junit.jupiter.engine.execution.MethodInvocation.proceed(MethodInvocation.java:60)
       org.junit.jupiter.engine.execution.InvocationInterceptorChain$ValidatingInvocation.proceed(InvocationInterceptorChain.java:131)
       [...]
  JUnit Jupiter:CommandLineInnerModelInnerUsageMessageSpecAnonymous1Test:autoUsageWidthDetectsTerminalInPseudoTerminalProcess()
    MethodSource [className = 'info_picocli.picocli.CommandLineInnerModelInnerUsageMessageSpecAnonymous1Test', methodName = 'autoUsageWidthDetectsTerminalInPseudoTerminalProcess', methodParameterTypes = '']
    => java.lang.NullPointerException
       java.base@25.0.3/java.util.Objects.requireNonNull(Objects.java:220)
       java.base@25.0.3/sun.nio.fs.UnixFileSystem.getPath(UnixFileSystem.java:257)
       java.base@25.0.3/java.nio.file.Path.of(Path.java:148)
       java.base@25.0.3/java.nio.file.Paths.get(Paths.java:69)
       info_picocli.picocli.CommandLineInnerModelInnerUsageMessageSpecAnonymous1Test.runChildProcessInPseudoTerminal(CommandLineInnerModelInnerUsageMessageSpecAnonymous1Test.java:63)
       info_picocli.picocli.CommandLineInnerModelInnerUsageMessageSpecAnonymous1Test.autoUsageWidthDetectsTerminalInPseudoTerminalProcess(CommandLineInnerModelInnerUsageMessageSpecAnonymous1Test.java:32)
       java.base@25.0.3/java.lang.reflect.Method.invoke(Method.java:565)
       org.junit.platform.commons.util.ReflectionUtils.invokeMethod(ReflectionUtils.java:727)
       org.junit.jupiter.engine.execution.MethodInvocation.proceed(MethodInvocation.java:60)
       org.junit.jupiter.engine.execution.InvocationInterceptorChain$ValidatingInvocation.proceed(InvocationInterceptorChain.java:131)
       [...]

Test run finished after 13 ms
[        17 containers found      ]
[         0 containers skipped    ]
[        17 containers started    ]
[         0 containers aborted    ]
[        17 containers successful ]
[         0 containers failed     ]
[        26 tests found           ]
[         0 tests skipped         ]
[        26 tests started         ]
[         0 tests aborted         ]
[        24 tests successful      ]
[         2 tests failed          ]


[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/info.picocli/picocli/4.7.6/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.
18 actionable tasks: 8 executed, 10 up-to-date

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':nativeTest'.
> Process 'command '/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/info.picocli/picocli/4.7.6/build/native/nativeTestCompile/info.picocli.picocli_tests-tests'' finished with non-zero exit value 1

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to generate a Build Scan (Powered by Develocity).
> Get more help at https://help.gradle.org.

BUILD FAILED in 1m 26s

> Task :tckTest FAILED

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/build/reports/problems/problems-report.html

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':tckTest'.
> Test for info.picocli:picocli:4.7.6 failed with exit code 1.

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to generate a Build Scan (Powered by Develocity).
> Get more help at https://help.gradle.org.

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD FAILED in 1m 27s
11 actionable tasks: 3 executed, 8 up-to-date
Finished stage 'native-image run' for 4.7.6 at 2026-05-12T10:18:35Z with exit code 1 after 88s
❌ ERROR: 'native-image run' for 4.7.6 failed with exit code 1.
FAILED[native-image run][4.7.6][GVM_TCK_NATIVE_IMAGE_MODE="current-defaults" GVM_TCK_LV="4.7.6" ./gradlew clean test -Pcoordinates="info.picocli:picocli:4.7.6"]
```

## native-image run on ubuntu-latest / JDK latest-ea / mode current-defaults

Failure kind: native-image run
OS: ubuntu-latest
JDK: latest-ea
Native-image mode: current-defaults
Reproducer: `GVM_TCK_NATIVE_IMAGE_MODE="current-defaults" GVM_TCK_LV="4.7.6" ./gradlew clean test -Pcoordinates="info.picocli:picocli:4.7.6"`
Runner log: https://github.com/oracle/graalvm-reachability-metadata/actions/runs/25727047580/job/75542581673
Last 300 lines of the log (excerpt truncated to fit the GitHub issue size limit):
```console
m/library-support.jar!/META-INF/native-image/com.oracle.svm/polyglot/native-image.properties+api=com.oracle.svm.polyglot.groovy.GroovyIndyInterfaceFeature' \
'-H:AddExports@jar:file:///opt/hostedtoolcache/graalvm-jdk-latest-ea_linux-x64_bin/25.0.0/x64/graalvm-25.1.0-dev+10.1/lib/svm/svm-guest.jar!/META-INF/native-image/org.graalvm.nativeimage.guest/native-image.properties+api=java.base/jdk.internal.misc=org.graalvm.nativeimage.guest' \
'-H:Features@jar:file:///opt/hostedtoolcache/graalvm-jdk-latest-ea_linux-x64_bin/25.0.0/x64/graalvm-25.1.0-dev+10.1/lib/svm/svm-guest.jar!/META-INF/native-image/org.graalvm.nativeimage.guest/native-image.properties+api=com.oracle.svm.guest.features.jdk.hosted.JavaNetHttpFeature' \
'-H:ClassInitialization@jar:file:///opt/hostedtoolcache/graalvm-jdk-latest-ea_linux-x64_bin/25.0.0/x64/graalvm-25.1.0-dev+10.1/lib/svm/svm-guest.jar!/META-INF/native-image/org.graalvm.nativeimage.guest/native-image.properties+api=sun.net.httpserver.simpleserver:run_time' \
'-H:ReachabilityMetadataResources@jar:file:///opt/hostedtoolcache/graalvm-jdk-latest-ea_linux-x64_bin/25.0.0/x64/graalvm-25.1.0-dev+10.1/lib/svm/svm-guest.jar!/META-INF/native-image/org.graalvm.nativeimage.guest/reachability-metadata.json+api=META-INF/native-image/org.graalvm.nativeimage.guest/reachability-metadata.json' \
-H:DriverWarningsCount=3
]
Warning: Option 'DynamicProxyConfigurationResources' is deprecated and might be removed in a future release: This can be caused by a proxy-config.json file in your META-INF directory. Consider including proxy configuration in the reflection section of reachability-metadata.json instead.. Please refer to the GraalVM release notes.
Warning: Option 'DynamicProxyConfigurationResources' is deprecated and might be removed in a future release: This can be caused by a proxy-config.json file in your META-INF directory. Consider including proxy configuration in the reflection section of reachability-metadata.json instead.. Please refer to the GraalVM release notes.
Warning: Option 'FallbackThreshold' is deprecated and might be removed in a future release: It no longer has any effect, and no replacement is available. Please refer to the GraalVM release notes.
========================================================================================================================
GraalVM Native Image: Generating 'info.picocli.picocli_tests-tests' (executable)...
========================================================================================================================
For detailed information and explanations on the build output, visit:
https://github.com/oracle/graal/blob/master/docs/reference-manual/native-image/BuildOutput.md
------------------------------------------------------------------------------------------------------------------------
[1/8] Initializing...                                                                                   (6.6s @ 0.18GiB)
 Builder configuration:
 - Java version: 25.0.2+10-LTS, vendor version: Oracle GraalVM 25.1.0-dev+10.1
 - Graal compiler: optimization level: b, target machine: x86-64-v3, PGO: off
 - C compiler: gcc (linux, x86_64, 13.3.0)
 - Assertions: disabled, system assertions: disabled
 - 1 user-specific feature(s):
   - org.graalvm.junit.platform.JUnitPlatformFeature
 Image configuration:
 - Garbage collector: Serial GC (max heap size: 80% of RAM)
 - Assertions: disabled (class-specific config may apply), system assertions: disabled
------------------------------------------------------------------------------------------------------------------------
 1 experimental option(s) unlocked:
 - '-H:BuildOutputJSONFile' (origin(s): command line)
------------------------------------------------------------------------------------------------------------------------
Build resources:
 - 13.27GiB of memory (85% of system memory because $CI set to 'true')
 - 4 thread(s) (100.0% of 4 available processor(s), determined at start)
[junit-platform-native] Running in 'test listener' mode using files matching pattern [junit-platform-unique-ids*] found in folder [/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/info.picocli/picocli/4.7.6/build/test-results/test/testlist] and its subfolders.
[2/8] Performing analysis...  [********]                                                               (34.8s @ 0.85GiB)
    8,553 types,   9,750 fields, and  38,441 methods found reachable
      342 types,     175 fields, and   1,023 methods registered for reflection
       60 types,      68 fields, and      52 methods registered for JNI access
      167 resource accesses registered with 262.53KiB total size
        4 native libraries: dl, pthread, rt, z
[3/8] Building universe...                                                                              (4.1s @ 0.97GiB)
[4/8] Parsing methods...      [**]                                                                      (4.5s @ 0.64GiB)
[5/8] Inlining methods...     [***]                                                                     (2.2s @ 0.68GiB)
[6/8] Compiling methods...    [****]                                                                   (20.3s @ 0.70GiB)
[7/8] Laying out methods...   [**]                                                                      (2.9s @ 0.86GiB)
[8/8] Creating image...       [**]                                                                      (3.4s @ 0.98GiB)
   8.97MiB (34.20%) for code area:    23,728 compilation units
  13.44MiB (51.22%) for image heap:  224,594 objects and 24 resources
   3.82MiB (14.58%) for other data
  26.24MiB in total image size, 24.19MiB in total file size
------------------------------------------------------------------------------------------------------------------------
Top 10 origins of code area:                                Top 10 object types in image heap:
   0.98MiB o.g.n.runtime/com.oracle.svm.core.code              2.84MiB byte[] for code metadata
 721.54KiB java.base/java.util                                 2.61MiB byte[] for string data
 524.29KiB picocli                                             1.40MiB java.base/java.lang.String
 408.42KiB java.base/jdk.internal.classfile.impl               1.02MiB java.base/java.lang.Class
 345.27KiB java.base/java.lang                               935.48KiB o.g.n.runtime/c.o.s.core.hub.DynamicHubCompanion
 275.97KiB java.base/java.text                               576.34KiB byte[] for general heap data
 243.18KiB java.base/java.util.concurrent                    356.50KiB java.base/java.util.HashMap$Node
 224.62KiB java.base/com.sun.crypto.provider                 349.53KiB byte[] for reflection metadata
 223.43KiB org.assertj.core.internal.bytebuddy.jar.asm       331.64KiB o.g.n.runtime/c.o.s.c.h.ImageReflectionMetadata
 163.84KiB java.base/java.io                                 271.38KiB java.base/java.lang.Object[]
   4.94MiB for 313 more packages                               2.81MiB for 2075 more object types
                            Use '--emit build-report' to create a report with more details.
------------------------------------------------------------------------------------------------------------------------
Security report:
 - Binary includes Java deserialization.
 - CycloneDX SBOM with 18 component(s) is embedded in binary (1.00KiB). 41 type(s) could not be associated to a component.
 - Advanced obfuscation not enabled; enable with '-H:AdvancedObfuscation=""' (experimental support).
------------------------------------------------------------------------------------------------------------------------
Recommendations:
 G1GC: Use the G1 GC ('--gc=G1') for improved latency and throughput.
 PGO:  Use Profile-Guided Optimizations ('--pgo') for improved throughput.
 FUTR: Use '--future-defaults=all' to prepare for future releases.
 HEAP: Set max heap for improved and more predictable memory usage.
 CPU:  Enable more CPU features with '-march=native' for improved performance.
------------------------------------------------------------------------------------------------------------------------
                       5.2s (6.3% of total time) in 742 GCs | Peak RSS: 1.58GiB | CPU load: 3.63
------------------------------------------------------------------------------------------------------------------------
Build artifacts:
 /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/info.picocli/picocli/4.7.6/build/native/nativeTestCompile/info.picocli.picocli_tests-tests (executable, 24.19MiB)
 /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/info.picocli/picocli/4.7.6/build/native/nativeTestCompile/libmanagement_ext.so (jdk_library, 32.23KiB)
 /tmp/native-image-build-output.json (build_info, 1.02KiB)
========================================================================================================================
Finished generating 'info.picocli.picocli_tests-tests' in 1m 21s.

The build process encountered 6 warnings.

[native-image-plugin] Native Image written to: /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/info.picocli/picocli/4.7.6/build/native/nativeTestCompile

> Task :nativeTest FAILED
[junit-platform-native] WARNING: -djunit.platform.listeners.uid.tracking.output.dir not specified, trying to find test-ids on default Gradle/Maven locations. As this is a fallback mode, it could take a while. This should only happen if you are running tests executable manually and you didn't pass uid output directory with -djunit.platform.listeners.uid.tracking.output.dir=<path>.
[junit-platform-native] WARNING: Using test-ids from default Gradle project location:/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/info.picocli/picocli/4.7.6/build/test-results/test/testlist
[junit-platform-native] Running in 'test listener' mode using files matching pattern [junit-platform-unique-ids*] found in folder [/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/info.picocli/picocli/4.7.6/build/test-results/test/testlist] and its subfolders.
JUnit Platform on Native Image - report
----------------------------------------

groovy.lang.CommandLineInnerDefaultFactoryTest > invokesGroovyClosureConstructorAndCallThroughDefaultFactory() SUCCESSFUL

groovy.lang.CommandLineInnerDefaultFactoryTest > createsPublicNoArgInstancesWithDefaultFactory() SUCCESSFUL

groovy.lang.CommandLineInnerDefaultFactoryTest > createsInstancesWithPrivateNoArgConstructors() SUCCESSFUL

groovy.lang.CommandLineInnerDefaultFactoryTest > invokesDeclaredConstructorWhenPublicInstantiationFails() SUCCESSFUL

groovy.lang.CommandLineInnerDefaultFactoryTest > createsInstanceWhenPublicConstructorSucceedsOnDeclaredRetry() SUCCESSFUL

info_picocli.picocli.AutoCompleteInnerAppTest > generatesCompletionScriptFromNamedCommandAndFactoryClasses() SUCCESSFUL

info_picocli.picocli.CommandLineInnerBuiltInInnerClassConverterTest > convertsFullyQualifiedClassNameOptionToClass() SUCCESSFUL

info_picocli.picocli.CommandLineInnerBuiltInInnerISO8601TimeConverterTest > convertsIso8601TimeOptionsToSqlTime() SUCCESSFUL

info_picocli.picocli.CommandLineInnerBuiltInInnerReflectionConverterTest > convertsBuiltInTypesUsingReflectionBackedConverters() SUCCESSFUL

info_picocli.picocli.CommandLineInnerHelpInnerAnsiTest > autoAnsiEvaluatesTerminalWhenTtyModeIsRequested() SUCCESSFUL

info_picocli.picocli.CommandLineInnerHelpInnerAnsiTest > autoAnsiEvaluatesConsoleTerminalInPseudoTerminalProcess() FAILED

info_picocli.picocli.CommandLineInnerInterpreterTest > expandsArrayOptionValues() SUCCESSFUL

info_picocli.picocli.CommandLineInnerInterpreterTest > togglesOptionalBooleanFromInitialValue() SUCCESSFUL

info_picocli.picocli.CommandLineInnerInterpreterTest > readsInteractivePasswordWithoutEcho() SUCCESSFUL

info_picocli.picocli.CommandLineInnerModelInnerArgGroupSpecTest > appendsRepeatedArgGroupInstancesToArrayField() SUCCESSFUL

info_picocli.picocli.CommandLineInnerModelInnerCommandSpecTest > resourceBundleBaseNameLoadsBundleAndUpdatesCommandArguments() SUCCESSFUL

info_picocli.picocli.CommandLineInnerModelInnerCommandUserObjectTest > populateSpecCreatesProxyForAnnotatedInterfaceCommand() SUCCESSFUL

info_picocli.picocli.CommandLineInnerModelInnerMessagesTest > loadsResourceBundleByBaseName() SUCCESSFUL

info_picocli.picocli.CommandLineInnerModelInnerMessagesTest > acceptsExistingResourceBundle() SUCCESSFUL

info_picocli.picocli.CommandLineInnerModelInnerMethodBindingTest > parseArgsInvokesAnnotatedSetterMethods() SUCCESSFUL

info_picocli.picocli.CommandLineInnerModelInnerUsageMessageSpecAnonymous1Test > autoUsageWidthDetectsTerminalInPseudoTerminalProcess() FAILED

info_picocli.picocli.CommandLineInnerPropertiesDefaultProviderTest > loadsDefaultsFromClasspathResourceWhenUserHomeFileIsAbsent(Path) SUCCESSFUL

info_picocli.picocli.CommandLineTest > executesStaticCommandMethod() SUCCESSFUL

info_picocli.picocli.CommandLineTest > mapsOptionalMethodOptionsToEmptyAndPresentValues() SUCCESSFUL

info_picocli.picocli.CommandLineTest > executesInstanceCommandMethodAsSubcommand() SUCCESSFUL

info_picocli.picocli.CommandLineTest > findsPublicInheritedAndDeclaredCommandMethods() SUCCESSFUL


Failures (2):
  JUnit Jupiter:CommandLineInnerHelpInnerAnsiTest:autoAnsiEvaluatesConsoleTerminalInPseudoTerminalProcess()
    MethodSource [className = 'info_picocli.picocli.CommandLineInnerHelpInnerAnsiTest', methodName = 'autoAnsiEvaluatesConsoleTerminalInPseudoTerminalProcess', methodParameterTypes = '']
    => java.lang.NullPointerException
       java.base@25.0.2/java.util.Objects.requireNonNull(Objects.java:220)
       java.base@25.0.2/sun.nio.fs.UnixFileSystem.getPath(UnixFileSystem.java:257)
       java.base@25.0.2/java.nio.file.Path.of(Path.java:148)
       java.base@25.0.2/java.nio.file.Paths.get(Paths.java:69)
       info_picocli.picocli.CommandLineInnerHelpInnerAnsiTest.runChildProcessInPseudoTerminal(CommandLineInnerHelpInnerAnsiTest.java:77)
       info_picocli.picocli.CommandLineInnerHelpInnerAnsiTest.autoAnsiEvaluatesConsoleTerminalInPseudoTerminalProcess(CommandLineInnerHelpInnerAnsiTest.java:48)
       java.base@25.0.2/java.lang.reflect.Method.invoke(Method.java:565)
       org.junit.platform.commons.util.ReflectionUtils.invokeMethod(ReflectionUtils.java:727)
       org.junit.jupiter.engine.execution.MethodInvocation.proceed(MethodInvocation.java:60)
       org.junit.jupiter.engine.execution.InvocationInterceptorChain$ValidatingInvocation.proceed(InvocationInterceptorChain.java:131)
       [...]
  JUnit Jupiter:CommandLineInnerModelInnerUsageMessageSpecAnonymous1Test:autoUsageWidthDetectsTerminalInPseudoTerminalProcess()
    MethodSource [className = 'info_picocli.picocli.CommandLineInnerModelInnerUsageMessageSpecAnonymous1Test', methodName = 'autoUsageWidthDetectsTerminalInPseudoTerminalProcess', methodParameterTypes = '']
    => java.lang.NullPointerException
       java.base@25.0.2/java.util.Objects.requireNonNull(Objects.java:220)
       java.base@25.0.2/sun.nio.fs.UnixFileSystem.getPath(UnixFileSystem.java:257)
       java.base@25.0.2/java.nio.file.Path.of(Path.java:148)
       java.base@25.0.2/java.nio.file.Paths.get(Paths.java:69)
       info_picocli.picocli.CommandLineInnerModelInnerUsageMessageSpecAnonymous1Test.runChildProcessInPseudoTerminal(CommandLineInnerModelInnerUsageMessageSpecAnonymous1Test.java:63)
       info_picocli.picocli.CommandLineInnerModelInnerUsageMessageSpecAnonymous1Test.autoUsageWidthDetectsTerminalInPseudoTerminalProcess(CommandLineInnerModelInnerUsageMessageSpecAnonymous1Test.java:32)
       java.base@25.0.2/java.lang.reflect.Method.invoke(Method.java:565)
       org.junit.platform.commons.util.ReflectionUtils.invokeMethod(ReflectionUtils.java:727)
       org.junit.jupiter.engine.execution.MethodInvocation.proceed(MethodInvocation.java:60)
       org.junit.jupiter.engine.execution.InvocationInterceptorChain$ValidatingInvocation.proceed(InvocationInterceptorChain.java:131)
       [...]

Test run finished after 16 ms
[        17 containers found      ]
[         0 containers skipped    ]
[        17 containers started    ]
[         0 containers aborted    ]
[        17 containers successful ]
[         0 containers failed     ]
[        26 tests found           ]
[         0 tests skipped         ]
[        26 tests started         ]
[         0 tests aborted         ]
[        24 tests successful      ]
[         2 tests failed          ]


[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/info.picocli/picocli/4.7.6/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.
18 actionable tasks: 8 executed, 10 up-to-date
WARNING: A terminally deprecated method in sun.misc.Unsafe has been called
WARNING: sun.misc.Unsafe::objectFieldOffset has been called by info_picocli.picocli.CommandLineInnerInterpreterTest (file:/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/info.picocli/picocli/4.7.6/build/native/nativeTestCompile/info.picocli.picocli_tests-tests)
WARNING: Please consider reporting this to the maintainers of class info_picocli.picocli.CommandLineInnerInterpreterTest
WARNING: sun.misc.Unsafe::objectFieldOffset will be removed in a future release

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':nativeTest'.
> Process 'command '/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/info.picocli/picocli/4.7.6/build/native/nativeTestCompile/info.picocli.picocli_tests-tests'' finished with non-zero exit value 1

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to generate a Build Scan (Powered by Develocity).
> Get more help at https://help.gradle.org.

BUILD FAILED in 1m 25s

> Task :tckTest FAILED

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/build/reports/problems/problems-report.html

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':tckTest'.
> Test for info.picocli:picocli:4.7.6 failed with exit code 1.

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to generate a Build Scan (Powered by Develocity).
> Get more help at https://help.gradle.org.

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD FAILED in 1m 27s
11 actionable tasks: 3 executed, 8 up-to-date
Finished stage 'native-image run' for 4.7.6 at 2026-05-12T10:18:21Z with exit code 1 after 88s
❌ ERROR: 'native-image run' for 4.7.6 failed with exit code 1.
FAILED[native-image run][4.7.6][GVM_TCK_NATIVE_IMAGE_MODE="current-defaults" GVM_TCK_LV="4.7.6" ./gradlew clean test -Pcoordinates="info.picocli:picocli:4.7.6"]
```

## native-image run on ubuntu-latest / JDK latest-ea / mode future-defaults-all

Failure kind: native-image run
OS: ubuntu-latest
JDK: latest-ea
Native-image mode: future-defaults-all
Reproducer: `GVM_TCK_NATIVE_IMAGE_MODE="future-defaults-all" GVM_TCK_LV="4.7.6" ./gradlew clean test -Pcoordinates="info.picocli:picocli:4.7.6"`
Runner log: https://github.com/oracle/graalvm-reachability-metadata/actions/runs/25727047580/job/75542581623
Last 300 lines of the log (excerpt truncated to fit the GitHub issue size limit):
```console
ea_linux-x64_bin/25.0.0/x64/graalvm-25.1.0-dev+10.1/lib/svm/library-support.jar!/META-INF/native-image/com.oracle.svm/polyglot/native-image.properties+api=com.oracle.svm.polyglot.groovy.GroovyIndyInterfaceFeature' \
'-H:AddExports@jar:file:///opt/hostedtoolcache/graalvm-jdk-latest-ea_linux-x64_bin/25.0.0/x64/graalvm-25.1.0-dev+10.1/lib/svm/svm-guest.jar!/META-INF/native-image/org.graalvm.nativeimage.guest/native-image.properties+api=java.base/jdk.internal.misc=org.graalvm.nativeimage.guest' \
'-H:Features@jar:file:///opt/hostedtoolcache/graalvm-jdk-latest-ea_linux-x64_bin/25.0.0/x64/graalvm-25.1.0-dev+10.1/lib/svm/svm-guest.jar!/META-INF/native-image/org.graalvm.nativeimage.guest/native-image.properties+api=com.oracle.svm.guest.features.jdk.hosted.JavaNetHttpFeature' \
'-H:ClassInitialization@jar:file:///opt/hostedtoolcache/graalvm-jdk-latest-ea_linux-x64_bin/25.0.0/x64/graalvm-25.1.0-dev+10.1/lib/svm/svm-guest.jar!/META-INF/native-image/org.graalvm.nativeimage.guest/native-image.properties+api=sun.net.httpserver.simpleserver:run_time' \
'-H:ReachabilityMetadataResources@jar:file:///opt/hostedtoolcache/graalvm-jdk-latest-ea_linux-x64_bin/25.0.0/x64/graalvm-25.1.0-dev+10.1/lib/svm/svm-guest.jar!/META-INF/native-image/org.graalvm.nativeimage.guest/reachability-metadata.json+api=META-INF/native-image/org.graalvm.nativeimage.guest/reachability-metadata.json' \
-H:DriverWarningsCount=3
]
Warning: Option 'DynamicProxyConfigurationResources' is deprecated and might be removed in a future release: This can be caused by a proxy-config.json file in your META-INF directory. Consider including proxy configuration in the reflection section of reachability-metadata.json instead.. Please refer to the GraalVM release notes.
Warning: Option 'DynamicProxyConfigurationResources' is deprecated and might be removed in a future release: This can be caused by a proxy-config.json file in your META-INF directory. Consider including proxy configuration in the reflection section of reachability-metadata.json instead.. Please refer to the GraalVM release notes.
Warning: Option 'FallbackThreshold' is deprecated and might be removed in a future release: It no longer has any effect, and no replacement is available. Please refer to the GraalVM release notes.
========================================================================================================================
GraalVM Native Image: Generating 'info.picocli.picocli_tests-tests' (executable)...
========================================================================================================================
For detailed information and explanations on the build output, visit:
https://github.com/oracle/graal/blob/master/docs/reference-manual/native-image/BuildOutput.md
------------------------------------------------------------------------------------------------------------------------
[1/8] Initializing...                                                                                   (6.3s @ 0.18GiB)
 Builder configuration:
 - Java version: 25.0.2+10-LTS, vendor version: Oracle GraalVM 25.1.0-dev+10.1
 - Graal compiler: optimization level: b, target machine: x86-64-v3, PGO: off
 - C compiler: gcc (linux, x86_64, 13.3.0)
 - Assertions: disabled, system assertions: disabled
 - 1 user-specific feature(s):
   - org.graalvm.junit.platform.JUnitPlatformFeature
 Image configuration:
 - Garbage collector: Serial GC (max heap size: 80% of RAM)
 - Assertions: disabled (class-specific config may apply), system assertions: disabled
------------------------------------------------------------------------------------------------------------------------
 1 experimental option(s) unlocked:
 - '-H:BuildOutputJSONFile' (origin(s): command line)
------------------------------------------------------------------------------------------------------------------------
Build resources:
 - 13.27GiB of memory (85% of system memory because $CI set to 'true')
 - 4 thread(s) (100.0% of 4 available processor(s), determined at start)
[junit-platform-native] Running in 'test listener' mode using files matching pattern [junit-platform-unique-ids*] found in folder [/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/info.picocli/picocli/4.7.6/build/test-results/test/testlist] and its subfolders.
[2/8] Performing analysis...  [*****]                                                                  (34.7s @ 0.92GiB)
    9,088 types,  10,546 fields, and  41,456 methods found reachable
      470 types,     175 fields, and   1,261 methods registered for reflection
       60 types,      68 fields, and      52 methods registered for JNI access
      169 resource accesses registered with 262.67KiB total size
        4 native libraries: dl, pthread, rt, z
[3/8] Building universe...                                                                              (4.0s @ 1.05GiB)
[4/8] Parsing methods...      [**]                                                                      (4.4s @ 0.74GiB)
[5/8] Inlining methods...     [***]                                                                     (2.0s @ 0.81GiB)
[6/8] Compiling methods...    [****]                                                                   (20.3s @ 0.85GiB)
[7/8] Laying out methods...   [**]                                                                      (3.5s @ 1.03GiB)
[8/8] Creating image...       [**]                                                                      (3.8s @ 1.16GiB)
  10.09MiB (33.75%) for code area:    26,043 compilation units
  15.75MiB (52.66%) for image heap:  263,695 objects and 26 resources
   4.07MiB (13.60%) for other data
  29.91MiB in total image size, 27.69MiB in total file size
------------------------------------------------------------------------------------------------------------------------
Top 10 origins of code area:                                Top 10 object types in image heap:
   1.11MiB o.g.n.runtime/com.oracle.svm.core.code              3.19MiB byte[] for code metadata
 728.57KiB java.base/java.util                                 2.91MiB byte[] for string data
 524.43KiB picocli                                             1.54MiB java.base/java.lang.String
 408.39KiB java.base/jdk.internal.classfile.impl               1.09MiB java.base/java.lang.Class
 354.14KiB java.base/java.lang                               994.00KiB o.g.n.runtime/c.o.s.core.hub.DynamicHubCompanion
 278.79KiB java.base/java.text                               922.45KiB byte[] for general heap data
 267.07KiB java.base/com.sun.crypto.provider                 450.38KiB heap alignment
 243.30KiB java.base/java.util.concurrent                    417.43KiB byte[] for reflection metadata
 223.43KiB org.assertj.core.internal.bytebuddy.jar.asm       373.05KiB java.base/java.lang.Object[]
 192.84KiB java.base/sun.security.provider                   359.28KiB java.base/java.util.HashMap$Node
   5.84MiB for 324 more packages                               3.58MiB for 2228 more object types
                            Use '--emit build-report' to create a report with more details.
------------------------------------------------------------------------------------------------------------------------
Security report:
 - Binary includes Java deserialization.
 - CycloneDX SBOM with 18 component(s) is embedded in binary (1.01KiB). 41 type(s) could not be associated to a component.
 - Advanced obfuscation not enabled; enable with '-H:AdvancedObfuscation=""' (experimental support).
------------------------------------------------------------------------------------------------------------------------
Recommendations:
 G1GC: Use the G1 GC ('--gc=G1') for improved latency and throughput.
 PGO:  Use Profile-Guided Optimizations ('--pgo') for improved throughput.
 HEAP: Set max heap for improved and more predictable memory usage.
 CPU:  Enable more CPU features with '-march=native' for improved performance.
------------------------------------------------------------------------------------------------------------------------
                       6.3s (7.7% of total time) in 801 GCs | Peak RSS: 1.79GiB | CPU load: 3.61
------------------------------------------------------------------------------------------------------------------------
Build artifacts:
 /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/info.picocli/picocli/4.7.6/build/native/nativeTestCompile/info.picocli.picocli_tests-tests (executable, 27.69MiB)
 /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/info.picocli/picocli/4.7.6/build/native/nativeTestCompile/libmanagement_ext.so (jdk_library, 32.23KiB)
 /tmp/native-image-build-output.json (build_info, 1.03KiB)
========================================================================================================================
Finished generating 'info.picocli.picocli_tests-tests' in 1m 21s.

The build process encountered 6 warnings.

[native-image-plugin] Native Image written to: /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/info.picocli/picocli/4.7.6/build/native/nativeTestCompile

> Task :nativeTest FAILED
[junit-platform-native] WARNING: -djunit.platform.listeners.uid.tracking.output.dir not specified, trying to find test-ids on default Gradle/Maven locations. As this is a fallback mode, it could take a while. This should only happen if you are running tests executable manually and you didn't pass uid output directory with -djunit.platform.listeners.uid.tracking.output.dir=<path>.
[junit-platform-native] WARNING: Using test-ids from default Gradle project location:/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/info.picocli/picocli/4.7.6/build/test-results/test/testlist
[junit-platform-native] Running in 'test listener' mode using files matching pattern [junit-platform-unique-ids*] found in folder [/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/info.picocli/picocli/4.7.6/build/test-results/test/testlist] and its subfolders.
JUnit Platform on Native Image - report
----------------------------------------

groovy.lang.CommandLineInnerDefaultFactoryTest > invokesGroovyClosureConstructorAndCallThroughDefaultFactory() SUCCESSFUL

groovy.lang.CommandLineInnerDefaultFactoryTest > createsPublicNoArgInstancesWithDefaultFactory() SUCCESSFUL

groovy.lang.CommandLineInnerDefaultFactoryTest > createsInstancesWithPrivateNoArgConstructors() SUCCESSFUL

groovy.lang.CommandLineInnerDefaultFactoryTest > invokesDeclaredConstructorWhenPublicInstantiationFails() SUCCESSFUL

groovy.lang.CommandLineInnerDefaultFactoryTest > createsInstanceWhenPublicConstructorSucceedsOnDeclaredRetry() SUCCESSFUL

info_picocli.picocli.AutoCompleteInnerAppTest > generatesCompletionScriptFromNamedCommandAndFactoryClasses() SUCCESSFUL

info_picocli.picocli.CommandLineInnerBuiltInInnerClassConverterTest > convertsFullyQualifiedClassNameOptionToClass() SUCCESSFUL

info_picocli.picocli.CommandLineInnerBuiltInInnerISO8601TimeConverterTest > convertsIso8601TimeOptionsToSqlTime() SUCCESSFUL

info_picocli.picocli.CommandLineInnerBuiltInInnerReflectionConverterTest > convertsBuiltInTypesUsingReflectionBackedConverters() SUCCESSFUL

info_picocli.picocli.CommandLineInnerHelpInnerAnsiTest > autoAnsiEvaluatesTerminalWhenTtyModeIsRequested() SUCCESSFUL

info_picocli.picocli.CommandLineInnerHelpInnerAnsiTest > autoAnsiEvaluatesConsoleTerminalInPseudoTerminalProcess() FAILED

info_picocli.picocli.CommandLineInnerInterpreterTest > expandsArrayOptionValues() SUCCESSFUL

info_picocli.picocli.CommandLineInnerInterpreterTest > togglesOptionalBooleanFromInitialValue() SUCCESSFUL

info_picocli.picocli.CommandLineInnerInterpreterTest > readsInteractivePasswordWithoutEcho() SUCCESSFUL

info_picocli.picocli.CommandLineInnerModelInnerArgGroupSpecTest > appendsRepeatedArgGroupInstancesToArrayField() SUCCESSFUL

info_picocli.picocli.CommandLineInnerModelInnerCommandSpecTest > resourceBundleBaseNameLoadsBundleAndUpdatesCommandArguments() SUCCESSFUL

info_picocli.picocli.CommandLineInnerModelInnerCommandUserObjectTest > populateSpecCreatesProxyForAnnotatedInterfaceCommand() SUCCESSFUL

info_picocli.picocli.CommandLineInnerModelInnerMessagesTest > loadsResourceBundleByBaseName() SUCCESSFUL

info_picocli.picocli.CommandLineInnerModelInnerMessagesTest > acceptsExistingResourceBundle() SUCCESSFUL

info_picocli.picocli.CommandLineInnerModelInnerMethodBindingTest > parseArgsInvokesAnnotatedSetterMethods() SUCCESSFUL

info_picocli.picocli.CommandLineInnerModelInnerUsageMessageSpecAnonymous1Test > autoUsageWidthDetectsTerminalInPseudoTerminalProcess() FAILED

info_picocli.picocli.CommandLineInnerPropertiesDefaultProviderTest > loadsDefaultsFromClasspathResourceWhenUserHomeFileIsAbsent(Path) SUCCESSFUL

info_picocli.picocli.CommandLineTest > executesStaticCommandMethod() SUCCESSFUL

info_picocli.picocli.CommandLineTest > mapsOptionalMethodOptionsToEmptyAndPresentValues() SUCCESSFUL

info_picocli.picocli.CommandLineTest > executesInstanceCommandMethodAsSubcommand() SUCCESSFUL

info_picocli.picocli.CommandLineTest > findsPublicInheritedAndDeclaredCommandMethods() SUCCESSFUL


Failures (2):
  JUnit Jupiter:CommandLineInnerHelpInnerAnsiTest:autoAnsiEvaluatesConsoleTerminalInPseudoTerminalProcess()
    MethodSource [className = 'info_picocli.picocli.CommandLineInnerHelpInnerAnsiTest', methodName = 'autoAnsiEvaluatesConsoleTerminalInPseudoTerminalProcess', methodParameterTypes = '']
    => java.lang.NullPointerException
       java.base@25.0.2/java.util.Objects.requireNonNull(Objects.java:220)
       java.base@25.0.2/sun.nio.fs.UnixFileSystem.getPath(UnixFileSystem.java:257)
       java.base@25.0.2/java.nio.file.Path.of(Path.java:148)
       java.base@25.0.2/java.nio.file.Paths.get(Paths.java:69)
       info_picocli.picocli.CommandLineInnerHelpInnerAnsiTest.runChildProcessInPseudoTerminal(CommandLineInnerHelpInnerAnsiTest.java:77)
       info_picocli.picocli.CommandLineInnerHelpInnerAnsiTest.autoAnsiEvaluatesConsoleTerminalInPseudoTerminalProcess(CommandLineInnerHelpInnerAnsiTest.java:48)
       java.base@25.0.2/java.lang.reflect.Method.invoke(Method.java:565)
       org.junit.platform.commons.util.ReflectionUtils.invokeMethod(ReflectionUtils.java:727)
       org.junit.jupiter.engine.execution.MethodInvocation.proceed(MethodInvocation.java:60)
       org.junit.jupiter.engine.execution.InvocationInterceptorChain$ValidatingInvocation.proceed(InvocationInterceptorChain.java:131)
       [...]
WARNING: A terminally deprecated method in sun.misc.Unsafe has been called
WARNING: sun.misc.Unsafe::objectFieldOffset has been called by info_picocli.picocli.CommandLineInnerInterpreterTest (file:/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/info.picocli/picocli/4.7.6/build/native/nativeTestCompile/info.picocli.picocli_tests-tests)
WARNING: Please consider reporting this to the maintainers of class info_picocli.picocli.CommandLineInnerInterpreterTest
WARNING: sun.misc.Unsafe::objectFieldOffset will be removed in a future release

  JUnit Jupiter:CommandLineInnerModelInnerUsageMessageSpecAnonymous1Test:autoUsageWidthDetectsTerminalInPseudoTerminalProcess()
    MethodSource [className = 'info_picocli.picocli.CommandLineInnerModelInnerUsageMessageSpecAnonymous1Test', methodName = 'autoUsageWidthDetectsTerminalInPseudoTerminalProcess', methodParameterTypes = '']
    => java.lang.NullPointerException
       java.base@25.0.2/java.util.Objects.requireNonNull(Objects.java:220)
       java.base@25.0.2/sun.nio.fs.UnixFileSystem.getPath(UnixFileSystem.java:257)
       java.base@25.0.2/java.nio.file.Path.of(Path.java:148)
       java.base@25.0.2/java.nio.file.Paths.get(Paths.java:69)
       info_picocli.picocli.CommandLineInnerModelInnerUsageMessageSpecAnonymous1Test.runChildProcessInPseudoTerminal(CommandLineInnerModelInnerUsageMessageSpecAnonymous1Test.java:63)
       info_picocli.picocli.CommandLineInnerModelInnerUsageMessageSpecAnonymous1Test.autoUsageWidthDetectsTerminalInPseudoTerminalProcess(CommandLineInnerModelInnerUsageMessageSpecAnonymous1Test.java:32)
       java.base@25.0.2/java.lang.reflect.Method.invoke(Method.java:565)
       org.junit.platform.commons.util.ReflectionUtils.invokeMethod(ReflectionUtils.java:727)
       org.junit.jupiter.engine.execution.MethodInvocation.proceed(MethodInvocation.java:60)
       org.junit.jupiter.engine.execution.InvocationInterceptorChain$ValidatingInvocation.proceed(InvocationInterceptorChain.java:131)
       [...]

Test run finished after 16 ms
[        17 containers found      ]
[         0 containers skipped    ]
[        17 containers started    ]
[         0 containers aborted    ]
[        17 containers successful ]
[         0 containers failed     ]
[        26 tests found           ]
[         0 tests skipped         ]
[        26 tests started         ]
[         0 tests aborted         ]
[        24 tests successful      ]
[         2 tests failed          ]


[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/info.picocli/picocli/4.7.6/build/reports/problems/problems-report.html
FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':nativeTest'.
> Process 'command '/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/info.picocli/picocli/4.7.6/build/native/nativeTestCompile/info.picocli.picocli_tests-tests'' finished with non-zero exit value 1

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to generate a Build Scan (Powered by Develocity).
> Get more help at https://help.gradle.org.

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD FAILED in 1m 25s
18 actionable tasks: 8 executed, 10 up-to-date

> Task :tckTest FAILED

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/build/reports/problems/problems-report.html

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':tckTest'.
> Test for info.picocli:picocli:4.7.6 failed with exit code 1.

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to generate a Build Scan (Powered by Develocity).
> Get more help at https://help.gradle.org.

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD FAILED in 1m 27s
11 actionable tasks: 3 executed, 8 up-to-date
Finished stage 'native-image run' for 4.7.6 at 2026-05-12T10:17:59Z with exit code 1 after 87s
❌ ERROR: 'native-image run' for 4.7.6 failed with exit code 1.
FAILED[native-image run][4.7.6][GVM_TCK_NATIVE_IMAGE_MODE="future-defaults-all" GVM_TCK_LV="4.7.6" ./gradlew clean test -Pcoordinates="info.picocli:picocli:4.7.6"]
```

--- a/tests/src/info.picocli/picocli/4.7.6/build.gradle
+++ b/tests/src/info.picocli/picocli/4.7.6/build.gradle
@@ -4,15 +4,40 @@
  * You should have received a copy of the CC0 legalcode along with this
  * work. If not, see <http://creativecommons.org/publicdomain/zero/1.0/>.
  */
+import java.io.OutputStream
+
 plugins {
     id "org.graalvm.internal.tck"
 }
 
 String libraryVersion = tck.testedLibraryVersion.get()
+Provider<Directory> generatedTestRuntimePropertiesDir = layout.buildDirectory.dir("generated/test-runtime-properties")
 
 dependencies {
     testImplementation "info.picocli:picocli:$libraryVersion"
     testImplementation 'org.assertj:assertj-core:3.22.0'
+}
+
+tasks.register("generateNativeRuntimeProperties") {
+    outputs.dir(generatedTestRuntimePropertiesDir)
+    doLast {
+        File outputDir = generatedTestRuntimePropertiesDir.get().asFile
+        outputDir.mkdirs()
+
+        Properties properties = new Properties()
+        properties.setProperty("java.home", System.getenv("JAVA_HOME") ?: "")
+        properties.setProperty("java.class.path", sourceSets.test.runtimeClasspath.asPath)
+
+        File outputFile = new File(outputDir, "native-runtime.properties")
+        outputFile.withOutputStream { OutputStream outputStream ->
+            properties.store(outputStream, "Generated runtime properties for picocli native tests")
+        }
+    }
+}
+
+processTestResources {
+    dependsOn(tasks.named("generateNativeRuntimeProperties"))
+    from(generatedTestRuntimePropertiesDir)
 }
 
 graalvmNative {

--- a/tests/src/info.picocli/picocli/4.7.6/src/test/java/info_picocli/picocli/CommandLineInnerHelpInnerAnsiTest.java
+++ b/tests/src/info.picocli/picocli/4.7.6/src/test/java/info_picocli/picocli/CommandLineInnerHelpInnerAnsiTest.java
@@ -74,6 +74,7 @@ public class CommandLineInnerHelpInnerAnsiTest {
     }
 
     private static ProcessResult runChildProcessInPseudoTerminal() throws IOException, InterruptedException {
+        NativeRuntimePropertiesSupport.restoreMissingRuntimeProperties();
         Path javaExecutable = Paths.get(System.getProperty("java.home"), "bin", javaExecutableName());
         Path scriptExecutable = Paths.get("/usr/bin/script");
         Path typescript = Files.createTempFile("picocli-ansi-pty", ".log");
@@ -100,10 +101,14 @@ public class CommandLineInnerHelpInnerAnsiTest {
 
     private static List<String> childJavaCommand(Path javaExecutable) {
         List<String> command = new ArrayList<>();
+        String classPath = System.getProperty("java.class.path", "");
+        if (classPath.isBlank()) {
+            throw new IllegalStateException("Could not locate child Java classpath");
+        }
         command.add(javaExecutable.toString());
         command.addAll(jacocoAgentArguments());
         command.add("-cp");
-        command.add(System.getProperty("java.class.path"));
+        command.add(classPath);
         command.add(CommandLineInnerHelpInnerAnsiTest.class.getName());
         command.add(CHILD_PROCESS_ARGUMENT);
         return command;

--- a/tests/src/info.picocli/picocli/4.7.6/src/test/java/info_picocli/picocli/CommandLineInnerModelInnerUsageMessageSpecAnonymous1Test.java
+++ b/tests/src/info.picocli/picocli/4.7.6/src/test/java/info_picocli/picocli/CommandLineInnerModelInnerUsageMessageSpecAnonymous1Test.java
@@ -60,6 +60,7 @@ public class CommandLineInnerModelInnerUsageMessageSpecAnonymous1Test {
     }
 
     private static ProcessResult runChildProcessInPseudoTerminal() throws IOException, InterruptedException {
+        NativeRuntimePropertiesSupport.restoreMissingRuntimeProperties();
         Path javaExecutable = Paths.get(System.getProperty("java.home"), "bin", javaExecutableName());
         Path scriptExecutable = Paths.get("/usr/bin/script");
         Path typescript = Files.createTempFile("picocli-usage-width-pty", ".log");
@@ -86,10 +87,14 @@ public class CommandLineInnerModelInnerUsageMessageSpecAnonymous1Test {
 
     private static List<String> childJavaCommand(Path javaExecutable) {
         List<String> command = new ArrayList<>();
+        String classPath = System.getProperty("java.class.path", "");
+        if (classPath.isBlank()) {
+            throw new IllegalStateException("Could not locate child Java classpath");
+        }
         command.add(javaExecutable.toString());
         command.addAll(jacocoAgentArguments());
         command.add("-cp");
-        command.add(System.getProperty("java.class.path"));
+        command.add(classPath);
         command.add(CommandLineInnerModelInnerUsageMessageSpecAnonymous1Test.class.getName());
         command.add(CHILD_PROCESS_ARGUMENT);
         return command;

--- a/tests/src/info.picocli/picocli/4.7.6/src/test/java/info_picocli/picocli/NativeRuntimePropertiesSupport.java
+++ b/tests/src/info.picocli/picocli/4.7.6/src/test/java/info_picocli/picocli/NativeRuntimePropertiesSupport.java
@@ -1,0 +1,59 @@
+/*
+ * Copyright and related rights waived via CC0
+ *
+ * You should have received a copy of the CC0 legalcode along with this
+ * work. If not, see <http://creativecommons.org/publicdomain/zero/1.0/>.
+ */
+package info_picocli.picocli;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.Properties;
+
+final class NativeRuntimePropertiesSupport {
+    private NativeRuntimePropertiesSupport() {
+    }
+
+    static void restoreMissingRuntimeProperties() {
+        Properties embeddedRuntimeProperties = loadEmbeddedRuntimeProperties();
+
+        if (System.getProperty("java.home") == null || System.getProperty("java.home").isBlank()) {
+            String javaHome = firstNonBlank(
+                    embeddedRuntimeProperties.getProperty("java.home"),
+                    System.getenv("JAVA_HOME"),
+                    System.getenv("GRAALVM_HOME"));
+            if (!javaHome.isBlank()) {
+                System.setProperty("java.home", javaHome);
+            }
+        }
+
+        String currentClassPath = System.getProperty("java.class.path", "");
+        if (currentClassPath.isBlank()) {
+            String classPath = embeddedRuntimeProperties.getProperty("java.class.path", "");
+            if (!classPath.isBlank()) {
+                System.setProperty("java.class.path", classPath);
+            }
+        }
+    }
+
+    private static Properties loadEmbeddedRuntimeProperties() {
+        Properties properties = new Properties();
+        try (InputStream inputStream = NativeRuntimePropertiesSupport.class.getResourceAsStream("/native-runtime.properties")) {
+            if (inputStream != null) {
+                properties.load(inputStream);
+            }
+        } catch (IOException exception) {
+            throw new IllegalStateException("Could not load embedded native runtime properties", exception);
+        }
+        return properties;
+    }
+
+    private static String firstNonBlank(String... candidates) {
+        for (String candidate : candidates) {
+            if (candidate != null && !candidate.isBlank()) {
+                return candidate;
+            }
+        }
+        return "";
+    }
+}

